### PR TITLE
ztest: Fix unused variable compile error in shuffle function

### LINKED ISSUE #48918: ztest: tests: add CONFIG_ZTEST_SHUFFLE=y to tests/subsys/logging/log_benchmark/prj.conf cause build failure
**Describe the bug**
After updating the new ztest api (https://github.com/zephyrproject-rtos/zephyr/pull/48902), add CONFIG_ZTEST_SHUFFLE=y to tests/subsys/logging/log_benchmark/prj.conf, build failure occurred while testing this module.

The error shows here :
zephyrproject/zephyr/subsys/testsuite/ztest/src/ztest_new.c:496:13: error: unused variable ‘start_pos’ [-Werror=unused-variable]:
image

**To Reproduce**
1. sync this pr(https://github.com/zephyrproject-rtos/zephyr/pull/48902).
2. add CONFIG_ZTEST_SHUFFLE=y to tests/subsys/logging/log_benchmark/prj.conf
3. west build -p always -b frdm_k64f tests/subsys/logging/log_benchmark/

**Environment :**
OS: Linux,
Toolchain Zephyr SDK 0.14.2
Commit SHA or Version: 2748fa508508d4873248375cfc798dcae1461cb3

### PR DESCRIPTION
**Pull request by asemjonovs**
When CONFIG_ZTEST_SHUFFLE is enabled and ASSERTS are disabled
`start_pos` becomes an unused variable leading to a compile error.
Cleaned-up shuffling algorithm to not need a `start_pos` check.

Signed-off-by: Al Semjonovs <asemjonovs@google.com>

Fixes #48918

### CODE REVIEW

**Comment by asemjonovs** [comment]
Fixes https://github.com/zephyrproject-rtos/zephyr/issues/48918

**Review by yperess — DISMISSED** [review]

**Comment by galak** [comment]
CI issues fixed in https://github.com/zephyrproject-rtos/zephyr/pull/49193

File: subsys/testsuite/ztest/src/ztest_new.c [hunk 1]
```diff
@@ -549,17 +549,22 @@ struct ztest_unit_test *z_ztest_get_next_test(const char *suite, struct ztest_un
...
 	for (size_t i = 0; i < num_items; ++i) {
-		int pos = sys_rand32_get() % num_items;
-		const int start_pos = pos;
+		dest[i] = (void *)(start + (i * element_size));
+	}
 
-		/* Get the next valid position */
-		while (dest[pos] != NULL) {
-			pos = (pos + 1) % num_items;
-			__ASSERT_NO_MSG(pos != start_pos);
```
Comment (aaronemassey) [inline subsys/testsuite/ztest/src/ztest_new.c:559]: Do we not care about what this is asserting anymore?

File: subsys/testsuite/ztest/src/ztest_new.c [hunk 2]
```diff
@@ -549,17 +549,22 @@ struct ztest_unit_test *z_ztest_get_next_test(const char *suite, struct ztest_un
 #ifdef CONFIG_ZTEST_SHUFFLE
 static void z_ztest_shuffle(void *dest[], intptr_t start, size_t num_items, size_t element_size)
 {
+	void *tmp;
+
+	/* Initialize dest array */
 	for (size_t i = 0; i < num_items; ++i) {
-		int pos = sys_rand32_get() % num_items;
-		const int start_pos = pos;
```
Comment (aaronemassey) [inline subsys/testsuite/ztest/src/ztest_new.c:554]: As a quick and dirty alternative, we could also just add `__maybe_unused` with a comment about asserts not being defined if we want to keep the `__ASSERT_NO_MSG(pos != start_pos);`

**Review by aaronemassey — CHANGES_REQUESTED** [review]

File: subsys/testsuite/ztest/src/ztest_new.c (hunk 1 above)
Comment (asemjonovs) [reply subsys/testsuite/ztest/src/ztest_new.c:559]: This assert was to catch an unlikely scenario of getting stuck in an infinite loop where `dest` has no NULL pointers.

File: subsys/testsuite/ztest/src/ztest_new.c (hunk 2 above)
Comment (asemjonovs) [reply subsys/testsuite/ztest/src/ztest_new.c:554]: That was my original fix, but I wanted to revert to my original implementation of randomizing an array which gives a better distribution.

**Review by yperess — APPROVED** [review]

**Review by aaronemassey — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**